contracts/settling_game/modules/relics/Relics.cairo

```diff
@@ -97,36 +129,220 @@ func set_relic_holder{pedersen_ptr: HashBuiltin*, syscall_ptr: felt*, range_chec
         tempvar pedersen_ptr = pedersen_ptr;
     }
 
-    // Capture back if held by loser
-    // Check if loser has winners token
-    let (winners_relic_owner) = get_current_relic_holder(winner_token_id);
+    let (realms_address) = IModuleController.get_external_contract_address(
+        controller, ExternalContractIds.Realms
+    );
 
-    let (holds_relic) = uint256_eq(winners_relic_owner, loser_token_id);
+    let (winner_realm_data: RealmData) = IRealms.fetch_realm_data(realms_address, winner_token_id);
 
-    if (holds_relic == TRUE) {
-        _set_relic_holder(winner_token_id, winner_token_id);
-        return ();
+    let (loser_relics_len) = owned_relics_len.read(loser_token_id);
+
+    let (remove_relics_len) = loop_claim_order_relic(
+        0, realms_address, winner_realm_data, loser_token_id, loser_relics_len, 0, remove_relics
```

Comment:
This `remove_relics` is potentially already set to `[loser_token_id]` from line 112, so then an attempt to write into index 0 on line 178 would fail.